### ISSUE TITLE
So Many Issues!

### ISSUE THREAD

**Issue by shiv-says** [realityfabric/Tumblr-Batch-Block#14]
- First, Tumblr Batch Block stopped working all together. No matter what I did, it would not do anything. I would put URLS separated by commas into the box, press the button, and nothing would happen.
- This went on for half a year! I would periodically try to use it, and nothing would happen! I uninstalled it and reinstalled it many times, and nothing! It just wouldn't work!
- Then, finally, Tumblr Batch Block began to do something this time. It would start blocking URLS. But it went so fast that it skipped 90% of the names I had put into the box.  
- So I tried to put the same list in the box again, hoping this time, it would block more urls than last time. And when it came to a name that had already been blocked, Tumblr Batch Block crashed, stop working, and then froze my entire computer!
- I had to uninstall and reinstall both Chrome and Tumblr Batch Block several times to make it block more urls, and it still kept skipping a bunch of the URLS on the list.
- Then, finally, Batch Block crashed and now, no matter what I do, it will not block anyone or anything. Just like it hasn't for the last 6 months.
- Added bonus, even though it's not blocking anyone, it still seems to be running and slowing down my computer and making Google Chrome crash

I really hope this all gets fixed, because I rely on this to make myself feel safe on Tumblr. This used to be so fast, easy, and convenient, but now it's just making my computer hot without actually benefiting me in any way.

**Comment by shiv-says** [comment]
(If this sounds angry, I'm sorry. I'm not mad at you. I'm just frustrated with the situation. Google stops responding, freezes, crashes, and closes on me if I try to use Tumblr Batch Block, and Tumblr Batch Block has stopped blocking people, and it's all very annoying.)

**Comment by shiv-says** [comment]
It seems that having a long list or having URLS in the list that have already been blocked make it harder for Tumblr Batch Block to function, and so part of the problem is coming from that. It stops blocking people as soon as it finds a URL that is already blocked, or if it has been blocking people for a minute or so straight. (Or at least, that's what happened before it stopped blocking people all together.) And once it stops blocking people, it causes Google to freeze and then crash. 

I'm sorry I keep adding to this. I just want to make sure my description of the issue makes sense.

**Comment by shiv-says** [comment]
Another issue I forgot to mention: it took Batch Block a really long time to start blocking people. I would put in the list, press the button, and then have to wait several minutes before anything actually happened. And it seemed to keep pausing during the process of blocking people, sometimes for several minutes at a time. The process, when it actually worked for me, was very slow. And then when things actually were working, like I said, it would go fast and skip over a ton of URLs. Then pause again. I couldn't tell if it was working or not.

**Comment by realityfabric** [comment]
I'm sorry that you're having so many issues with TBB! I haven't updated it in over a year (I haven't used Tumblr in about that long) so Tumblr or Chrome must have changed something.

The last change I made back in July 2019 was to make blocking faster, but the unintended side effect was that people with large block lists ended up having to wait longer. The reason is that the extension reads the entire block list before starting, so that it can remove usernames that have already been blocked.

I ran a quick test just now and I don't have most of these issues, so I assume it has to do with the number of users being blocked or already blocked. This is definitely an issue, but it's hard to test for.

I'm going to revert the extension to the previous version and add a request for someone to take over development.

**Comment by realityfabric** [comment]
I am unable to update the extension because Google now requires me to pay them to use my developer account.

I posted a statement and request for developers to take over on Tumblr, if you would like to share it.

https://realityfabric.tumblr.com/post/625362654810243072/i-am-no-longer-developing-tbb
